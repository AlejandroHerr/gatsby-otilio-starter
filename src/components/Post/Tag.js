// @flow
import React from 'react';
import Link from 'gatsby-link';

import styles from './Post.module.scss';

const Tag = ({ tag }: {tag: string}) => (
  <Link className={styles.post_tags__tag} to={`/tags/${tag.toLowerCase()}`}>{tag}</Link>
);

export default Tag;
