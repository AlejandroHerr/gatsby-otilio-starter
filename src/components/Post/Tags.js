// @flow
import React from 'react';

import PostTag from './Tag';

import styles from './Post.module.scss';

const Tags = ({ tags }: { tags: Array<string>}) => (
  <aside className={styles.post_tags}>
    {tags.map(tag => <PostTag key={tag} tag={tag} />)}
  </aside>
);

export default Tags;
