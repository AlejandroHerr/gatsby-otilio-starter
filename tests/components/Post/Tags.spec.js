import React from 'react';
import { shallow } from 'enzyme';

import Tags from '../../../src/components/Post/Tags';
import Tag from '../../../src/components/Post/Tag';

const defaultProps = {
  tags: ['test_1', 'test_2'],
};

const setup = (props = {}) => {
  const tags = shallow(<Tags {...defaultProps} {...props} />);

  return {
    tags,
    tag: tags.find(Tag),
  };
};

describe('Post', () => {
  describe('Tags', () => {
    it('expect to render one Tag for every tag', () => {
      const { tags, tag } = setup();

      expect(tag).toHaveLength(defaultProps.tags.length);
      expect(tags).toMatchSnapshot();
    });
  });
});
