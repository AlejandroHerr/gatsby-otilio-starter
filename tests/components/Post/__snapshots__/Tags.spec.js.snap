// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Tags expect to render one Tag for every tag 1`] = `
<aside
  className="post_tags"
>
  <Tag
    key="test_1"
    tag="test_1"
  />
  <Tag
    key="test_2"
    tag="test_2"
  />
</aside>
`;
